massa-grpc/src/api.rs

```diff
@@ -0,0 +1,894 @@
+//! Copyright (c) 2022 MASSA LABS <info@massa.net>
+//! gRPC API for a massa-node
+use std::{collections::HashMap, error::Error, io::ErrorKind, pin::Pin, str::FromStr};
+
+use crate::config::GrpcConfig;
+use itertools::izip;
+use massa_consensus_exports::{ConsensusChannels, ConsensusController};
+use massa_execution_exports::ExecutionController;
+use massa_models::{
+    address::Address,
+    block::{BlockDeserializer, BlockDeserializerArgs, SecureShareBlock},
+    endorsement::{EndorsementDeserializer, SecureShareEndorsement},
+    error::ModelsError,
+    operation::{OperationDeserializer, SecureShareOperation},
+    secure_share::SecureShareDeserializer,
+    slot::Slot,
+    timeslots,
+};
+use massa_pool_exports::{PoolChannels, PoolController};
+use massa_pos_exports::SelectorController;
+use massa_proto::massa::api::v1::{self as grpc, grpc_server::GrpcServer, FILE_DESCRIPTOR_SET};
+use massa_serialization::{DeserializeError, Deserializer};
+
+use futures_util::{FutureExt, StreamExt};
+use massa_protocol_exports::ProtocolCommandSender;
+use massa_storage::Storage;
+use tokio::sync::oneshot;
+use tonic::codegen::{futures_core, CompressionEncoding};
+use tonic_web::GrpcWebLayer;
+use tracing::log::{error, info, warn};
+
+/// gRPC API content
+pub struct MassaService {
```

Comment:
rename to MassaGrpcService ?  `MassaService` it's too generic

Reply:
Naming is hard, why not. But IMHO, we should talk about It as we should split the BIG service into smaller one's by domain.